File: Storage/Java/app/src/main/java/com/naufalprakoso/javaimagegallery/MainActivity.java
package com.naufalprakoso.javaimagegallery;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.design.widget.FloatingActionButton;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.GridLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.Toolbar;
import android.util.Log;
import android.view.View;

import com.google.firebase.database.DataSnapshot;
import com.google.firebase.database.DatabaseError;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;
import com.google.firebase.database.ValueEventListener;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    private RecyclerView recyclerView;
    private ArrayList<Image> images;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        FloatingActionButton fab = findViewById(R.id.fab);
        recyclerView = findViewById(R.id.rv_image);

        fab.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(getApplicationContext(), AddImageActivity.class);
                startActivity(intent);
            }
        });

        recyclerView.setLayoutManager(new GridLayoutManager(this, 2));
        images = new ArrayList<>();
        DatabaseReference databaseRef = FirebaseDatabase
                .getInstance()
                .getReference("images");

        databaseRef.addValueEventListener(new ValueEventListener() {
            @Override
            public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
                images.clear();
                for (DataSnapshot snapshot : dataSnapshot.getChildren()){
                    Image image = snapshot.getValue(Image.class);
                    images.add(image);
                }

                recyclerView.setAdapter(new ImageAdapter(getApplicationContext(), images));
            }

            @Override
            public void onCancelled(@NonNull DatabaseError databaseError) {
                Log.d("MainActivity", databaseError.getMessage());
            }
        });
    }

}

File: Storage/Kotlin/app/src/main/java/com/naufalprakoso/kotlinfirebasestorage/MainActivity.kt
package com.naufalprakoso.kotlinfirebasestorage

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.GridLayoutManager
import com.google.firebase.database.DataSnapshot
import com.google.firebase.database.DatabaseError
import com.google.firebase.database.FirebaseDatabase
import com.google.firebase.database.ValueEventListener

import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.content_main.*
import org.jetbrains.anko.startActivity

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        setSupportActionBar(toolbar)

        fab.setOnClickListener {
            startActivity<AddImage>()
        }
        
        val databaseRef = FirebaseDatabase.getInstance().getReference("images")
	    val images: ArrayList<Image> = ArrayList()
	    rv_image.layoutManager = GridLayoutManager(this, 2)
	    
	    databaseRef.addValueEventListener(object : ValueEventListener {
		    override fun onCancelled(p0: DatabaseError) {
			    println("Info: ${p0.message}")
		    }
		
		    override fun onDataChange(p0: DataSnapshot) {
			    images.clear()
			    for (dataSnapshot in p0.children){
				    val getValue = dataSnapshot.getValue(Image::class.java)
				    getValue?.let { images.add(it) }
			    }
			    
			    rv_image.adapter = ImageAdapter(applicationContext, images)
		    }
	    })
    }
}
